File: backend/src/main/java/org/globe42/web/persons/PersonIdentityDTO.java
package org.globe42.web.persons;

import org.globe42.domain.Person;

/**
 * A DTO allowing to identify a person. Used in person list.
 * @author JB Nizet
 */
public final class PersonIdentityDTO {
    private final Long id;
    private final String firstName;
    private final String lastName;
    private final String nickName;
    private final String mediationCode;

    public PersonIdentityDTO(Person person) {
        this.id = person.getId();
        this.firstName = person.getFirstName();
        this.lastName = person.getLastName();
        this.nickName = person.getNickName();
        // we hide the mediation code if mediation is disabled
        this.mediationCode = person.isMediationEnabled() ? person.getMediationCode() : null;
    }

    public Long getId() {
        return id;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getNickName() {
        return nickName;
    }

    public String getMediationCode() {
        return mediationCode;
    }
}

File: backend/src/main/kotlin/org/globe42/web/persons/PersonIdentityDTO.kt
package org.globe42.web.persons

import org.globe42.domain.Person

/**
 * A DTO allowing to identify a person. Used in person list.
 * @author JB Nizet
 */
data class PersonIdentityDTO(
    val id: Long,
    val firstName: String,
    val lastName: String,
    val nickName: String?,
    val mediationCode: String?,
    val phoneNumber: String?,
    val email: String?
) {

    constructor(person: Person) : this(
        id = person.id!!,
        firstName = person.firstName,
        lastName = person.lastName,
        nickName = person.nickName,
        // we hide the mediation code if mediation is disabled
        mediationCode = if (person.mediationEnabled) person.mediationCode else null,
        phoneNumber = person.phoneNumber,
        email = person.email
    )
}
